Classes/Handler/FileUploadHandler.php

```diff
@@ -87,17 +89,29 @@ public function uploadFile(InterestRequestInterface $request): ResponseInterface
         $fileBaseName = $data['data']['name'];
 
         if ($storage->hasFileInFolder($fileBaseName, $downloadFolder)) {
-            if (!$data['overwrite']) {
-                return $responseFactory->createErrorResponse(
-                    ['status' => 'failure', 'message' => 'File already exists, no overwrite access given.'],
-                    401,
+            if ($data['data']['overwriteFlag'] === 'use') {
```

Comment:
Consider if there are other flags that should be introduced here.

use = use the file if it exists
overwrite = overwrite the file
fail = fail if the file exists
rename = rename (the uploaded file) by adding a number to the end of the name.